src/applications/hca/config/form.js

```diff
@@ -164,6 +166,14 @@ const formConfig = {
   },
   transformForSubmit: transform,
   introduction: IntroductionPage,
+  additionalRoutes: !environment.isProduction() && [
+    {
+      path: 'id-form',
+      component: IDPage,
+      pageKey: 'id-form',
+      depends: () => !hasSession(),
```

Comment:
This is not great... Normally the `depends` callback should only consider form data, and ideally it should be able to look at other slices of state (in this case, `isLoggedIn(state)`), but this is what I came up with without making too many substantial changes to the forms library.

Reply:
does this actually work? in my quick tests running this locally, this `depends` callback isn't getting fired. I'm not sure why it's not working. It looks like the depends callback gets injected into the pageList correctly. I'll dig into it further but let me know if you have any thoughts.

Reply:
It seemed to work for me. If you click "Start Application" when you're logged in, it should skip right over the ID form into the first actual form page. And clicking back on that page should go back to the intro. In what scenario is it not working on your end?

Reply:
Okay I'll try that. I was just curiously poking around and stuck a `debugger;` line in the callback but execution never froze, meaning that line was never actually reached. Not sure why. Will try again!

Reply:
Yeah this depends callback doesn't get fired, but things do work how you describe. At this point I'm not sure _why_ the `id-form` route is skipped for logged-in users, but it is.